nixos/modules/programs/ssh.nix

```diff
@@ -62,6 +62,29 @@ in
         '';
       };
 
+      # Allow DSA keys for now. (These were deprecated in OpenSSH 7.0.)
+      pubkeyAcceptedKeyTypes = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = [
+          "+ssh-dss"
+        ];
+        example = [ "ssh-ed25519" "ssh-rsa" ];
+        description = ''
+          Specifies the key types that will be used for public key authentication.
+        '';
+      };
+
+      hostKeyAlgorithms = mkOption {
+        type = types.nullOr (types.listOf types.str);
```

Comment:
The type of those can be just `types.listOf types.str`, then an empty list replaces the null case.